tester/src/test/scala/spinal/tester/scalatest/SpinalSimBmbExclusiveMonitorTester.scala

```diff
@@ -137,8 +137,10 @@ class SpinalSimBmbExclusiveMonitorTester extends SpinalAnyFunSuite{
             memory = v
           }
           dut.io.output.rsp.opcode #= Bmb.Rsp.Opcode.SUCCESS
-          dut.io.output.rsp.context #= transaction.context.asInstanceOf[BigInt]
-          dut.io.output.rsp.source #= transaction.source.asInstanceOf[BigInt]
+          val context = transaction.context.asInstanceOf[BigInt]
+          dut.io.output.rsp.context #= context
+          val source = transaction.source.asInstanceOf[BigInt]
+          dut.io.output.rsp.source #= source
```

Comment:
I am not sure, for some reason it was otherwise not resolving the type of the right hand side to `BigInt`. Any ideas of something else I could try there?

I also think once we have a way to automatically implement `SimEquiv` for any bundle (By also defining a struct that can be used in simulation), then there will be little reason to use `SimData`, whose use is the reason for the dynamic casting here. Do you agree?

Reply:
Hmm weird scala XD

right.